### PR TITLE
Add milestones counter to statscollector

### PR DESCRIPTION
**Pull request by tsvisabo**

### CODE REVIEW

File: tanglescope/common/iri.cpp [hunk 1]
```diff
@@ -17,7 +17,9 @@ std::shared_ptr<IRIMessage> payloadToMsg(std::string_view payload) {
     return std::make_shared<SNMessage>(actual);
   } else if (what == "lmhs") {
     return std::make_shared<LMHSMessage>(actual);
-  } else if (what == "rstat") {
+  } else if (what == "lmsi") {
+    return std::make_shared<LMSIMessage>(actual);
+  }else if (what == "rstat") {
```
Comment (th0br0) [inline tanglescope/common/iri.cpp]: `clang-format` please

File: tanglescope/common/iri.hpp [hunk 2]
```diff
@@ -174,6 +174,22 @@ class LMHSMessage : public IRIMessage {
   std::string _latestSolidMilestoneHash;
 };
 
+class LMSIMessage : public IRIMessage {
+ public:
```
Comment (th0br0) [inline tanglescope/common/iri.hpp:178]: you might want to delete the default constructor?

File: tanglescope/common/iri.hpp [hunk 3]
```diff
@@ -174,6 +174,22 @@ class LMHSMessage : public IRIMessage {
...
+  explicit LMSIMessage(std::string_view from) {
+    auto chunks = IRIMessage::chunks(from);
+    this->_latestSolidMilestoneIndex = std::stoull(std::move(chunks[1]));
+  }
+
+  inline IRIMessageType type() { return IRIMessageType::LMSI; }
+
+ public:
+  const uint64_t& latestSolidMilestoneIndex() { return this->_latestSolidMilestoneIndex; }
```
Comment (th0br0) [inline tanglescope/common/iri.hpp]: shouldn't return by reference here. a reference is just as large as the actual value anyway.

File: tanglescope/statscollector/statscollector.cpp [hunk 4]
```diff
@@ -115,11 +117,13 @@ void ZMQCollectorImpl::collect(uint32_t bundleConfirmationHistogramRange,
       rxcpp::observable<>::create<std::shared_ptr<iri::IRIMessage>>(
           [&](auto s) { zmqPublisher(std::move(s), _zmqURL); });
 
+  uint64_t lmsi = 0;
```
Comment (th0br0) [inline tanglescope/statscollector/statscollector.cpp]: add a comment to explain what `lmsi` stands for?

**Review by th0br0 — CHANGES_REQUESTED** [review]

File: tanglescope/common/iri.hpp (hunk 3 above)
Comment (tsvisabo) [reply tanglescope/common/iri.hpp]: was overlooking, thanks

File: tanglescope/statscollector/statscollector.cpp (hunk 4 above)
Comment (tsvisabo) [reply tanglescope/statscollector/statscollector.cpp]: yup

File: tanglescope/common/iri.hpp (hunk 2 above)
Comment (tsvisabo) [reply tanglescope/common/iri.hpp:178]: Defining a one parameter constructor will prevent the compiler from generating default one,
so no unintentional usage of that is possible

**Review by th0br0 — APPROVED** [review]

**Pull request merged**